keyboards/setta21/keymaps/salicylic/keymap.c

```diff
@@ -0,0 +1,201 @@
+          XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+      //|--------+--------+--------+--------|
+          RGB_SAD, RGB_SAI, XXXXXXX,         
+      //|--------+--------+--------+--------|
+          RGB_HUD, RGB_HUI, XXXXXXX, RGB_TOG,
+      //|--------+--------+--------+--------|
+          RGB_VAD, RGB_VAI, XXXXXXX,         
+      //|--------+--------+--------+--------|
+          _______,          _______, RGB_MOD 
+      //`-----------------------------------'
+  )
+};
+
+
+//A description for expressing the layer position in LED mode.
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _ARROW, _MACRO, _ADJUST);
+}
+
+int RGB_current_mode;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  if (record->event.pressed) {
+     #ifdef OLED_DRIVER_ENABLE
+        oled_timer = timer_read32();
+     #endif
+  }
+  switch (keycode) {
+    case SEND_SUM:
+      if (record->event.pressed) {
+        SEND_STRING("_SUM*");
+      }
+      break;
+    case SEND_AVE:
+      if (record->event.pressed) {
+        SEND_STRING("_AVERAGE*");
+      }
+      break;
+    case SEND_CIF:
+      if (record->event.pressed) {
+        SEND_STRING("_COUNTIF*");
+      }
+      break;
+    case SEND_MAX:
+      if (record->event.pressed) {
+        SEND_STRING("_MAX*");
+      }
+      break;
+    case SEND_MIN:
+      if (record->event.pressed) {
+        SEND_STRING("_MIN*");
+      }
+      break;
+    #ifdef RGBLIGHT_ENABLE
+      case RGB_MOD:
+          if (record->event.pressed) {
+            rgblight_mode(RGB_current_mode);
+            rgblight_step();
+            RGB_current_mode = rgblight_config.mode;
+          }
+        break;
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
+          }
+        break;
+    #endif
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
+
+#ifdef OLED_DRIVER_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) { return OLED_ROTATION_0; }
+
+
+void render_layer_state(void) {
+    oled_write_P(PSTR("LAYER: "), false);
+    oled_write_P(PSTR(" Arrow "), layer_state_is(_ARROW));
+    oled_write_P(PSTR(" Macro "), layer_state_is(_MACRO));
+}
+
+void render_keylock_status(uint8_t led_usb_state) {
+    oled_write_P(PSTR("NumLock"), led_usb_state & (1 << USB_LED_NUM_LOCK));
+    oled_write_P(PSTR("              "), false);
+}
+
+void render_layer_messages(void) {
+    oled_write_P(PSTR("Setta21                For Your Good Job. "), false);
+}
+
+
+void render_status(void) {
+    /* Show Keyboard Layout  */
+    render_layer_messages();
+    render_keylock_status(host_keyboard_leds());
+    render_layer_state();
+}
+
+void oled_task_user(void) {
+    if (timer_elapsed32(oled_timer) > 30000) {
+        oled_off();
+        return;
+    }
```

Comment:
Just a heads up, the OLED Driver does have a built in screen timeout.  And that may still work.

This implementation is mostly needed for boards that have a custom matrix that don't work properly, or for some splits, because it's not processed correctly by default. 

So I'm not sure that this is even needed for this board

Reply:
if it erased the description, it was deleted.
Thank you.